packages/react-scripts/bin/react-scripts.js

```diff
@@ -12,18 +12,17 @@
 
 var spawn = require('cross-spawn');
 var script = process.argv[2];
-var args = process.argv.slice(3);
+var getArgs = require('../scripts/utils/getSubscriptArgs');
 
 switch (script) {
   case 'build':
   case 'eject':
   case 'start':
   case 'test':
-    var result = spawn.sync(
-      'node',
-      [require.resolve('../scripts/' + script)].concat(args),
-      { stdio: 'inherit' }
-    );
+    var scriptFilename = require.resolve('../scripts/' + script);
+    var result = spawn.sync('node', getArgs(scriptFilename), {
+      stdio: 'inherit',
+    });
```

Comment:
right right right.  if desired, we can annotate if you think its fitting somewhere, or rename the fn's to be more communicative